packages/rest/src/rules/duplicate-parent-key.ts

```diff
@@ -0,0 +1,43 @@
+import { getKeyName, getServiceNamespace, isDefinedInNamespace, Model } from "@cadl-lang/compiler";
+import { createRule } from "@cadl-lang/lint";
+import { reportDiagnostic } from "../lib.js";
+import { getParentResource, getResourceTypeKey } from "../resource.js";
+
+export const duplicateParentKey = createRule({
+  name: "duplicate-parent-key",
```

Comment:
Does this make sense to be a linter rule?

Might need to write some when to use the linter vs pure validation but the way I was seeing it:
- if this is something that makes the cadl representation invalid it should just be a validation(There shouldn't a way to toggle it). 
- if this is a style question, best practice then it can be a linter

This does feel weird specially as we already have a validation for this in the parentResource decorator

Reply:
yeah I don't mean not include this change but just not use the linter engine for it.

Reply:
So you're suggesting to make it a plain function call in `onValidate` then?

Reply:
OK, I see that this is what you meant before.  The order that GitHub displayed the different comment threads made it hard to tell.

I chose to use the linter engine because it seems like we'll eventually want other rules but I can remove that if you think it's inappropriate for now.

Reply:
yeah I can definitely see more linter rules being added but I think we shouldn't have things like this as linter rule which could get enabled/disabled by other extensions

Reply:
Done.

Reply:
@daviwil can you say more about the cases when a child type is evaluated before a parent type?

Reply:
@bterlson A partner team ran into an issue related to this, issue #960 has all the details and a repro.  The repro no longer works because I added a mitigation in PR #966.